[ticket/15037] Make imagesets retina

hanakin opened the pull request:
Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)


* read the ticket for more info

PHPBB3-15037

https://tracker.phpbb.com/browse/PHPBB3-15037

hanakin commented:
!set Styles

hanakin commented:
some screens

<img width="138" alt="screen shot 2017-01-21 at 6 02 00 pm" src="https://cloud.githubusercontent.com/assets/73081/22178428/d1520b50-e003-11e6-9f4f-3bdf1f31815b.png">
<img width="180" alt="screen shot 2017-01-21 at 6 02 13 pm" src="https://cloud.githubusercontent.com/assets/73081/22178429/d1547430-e003-11e6-963f-a3e95b54f010.png">

JimMH commented:
Looks very good. Wouldn't it be better though to replace the existing images instead of using new ones and changing the CSS? Otherwise this might break custom styles.

Crizz0 commented:
He changed the file type to .png (was .gif). I thinkt this i neccessary for better quality? 

But why do you add this "-2x" to the filename?

hanakin commented:
@JimMH  I made two changes to the colors.css file of note.

1. I used PNG for 2 reasons

    1. PNG is the std format for all graphic editors
    2. GIF is a dead tech its not used anymore as its not scaler meaning it does not support scaling of the img as other formats do. You get artifacting and other weird issues sometimes.

I can try converting them to gifs after all the icons are good to go as they are all new from scratch.

<img width="568" alt="screen shot 2017-01-21 at 6 27 51 pm" src="https://cloud.githubusercontent.com/assets/73081/22178555/612fc886-e007-11e6-94a0-fc545822b818.png">


2. I also changed the names as well to remove the stupid fucking '_' these are highly problematic for text editors and image editors and makes it take 5 times longer to make changes as you can not chain with them. again I can rename them back easily if there are no issues found with any of the images

@Crizz0 its the way my sketch works it saves a normal version and a double sized version with that on the end I removed the normal versions I just did not remove the appending yet.

hanakin commented:
!set Wip

hanakin commented:
!set WIP

hanakin commented:
@JimMH here is the difference of png vs gif

<img width="237" alt="screenshot 2017-01-21 19 35 33" src="https://cloud.githubusercontent.com/assets/73081/22178849/e565955a-e010-11e6-8e35-06f4d12f02bb.png">

marc1706 commented:
Any reason the locks on stickies are further to the top of the image than on the other icons?

VSEphpbb commented:
No more GIFs. Lets stickw ith PNG. If anything I'd say use SVGs with PNG fallbacks ;)

hanakin commented:
@marc1706 the locks is due to the shape at the bottom being narrower and the icon floating. its this way on the current sticky icon for the same reason. 

@VSEphpbb SVGs will be used in 3.3 for sure problem is some aspects of core oes not support svg which will need fixed in 3.3. This is meant as a quick fix for 3.2 in the mean time. There is no need for a fallback anymore svg is supported in all browsers. Although not all features are supported across the board so careful design practices and the way we serve them need to be taken into account.

hanakin commented:
!unset WIP

marc1706 commented:
Changes look good to me. @Nicofuma want to have this in 3.2.1 or wait for 3.2.2 ?

Nicofuma commented:
3.2.2 if we have gifs (ie: same image names as in 3.2.0), otherwise 3.3 because it will break almost all 3.2 styles.

hanakin commented:
@Nicofuma gif is not possible

hanakin commented:
@Nicofuma I may have found a way to get gifs working but it means recreating all the images from scratch again and will take me a little so stby. here is one for an example
![topic-read](https://cloud.githubusercontent.com/assets/73081/22191483/1c709548-e0f9-11e6-8366-62f15125eb17.gif)

marc1706 commented:
I think that would be better, especially regarding BC.

hanakin commented:
Should have this up for review tonight just need to render unread versions

Nicofuma commented:
Thanks a lot

hanakin commented:
k should be all good to go pending tests here is an updated screen

<img width="186" alt="screenshot 2017-01-23 19 35 23" src="https://cloud.githubusercontent.com/assets/73081/22229033/309ea0be-e1a3-11e6-832e-c28ac1ff5605.png">

hanakin commented:
possible counter part to this

<img width="115" alt="screenshot 2017-01-23 20 50 18" src="https://cloud.githubusercontent.com/assets/73081/22230969/b85845d6-e1ae-11e6-8192-40b6090366c6.png">

VSEphpbb commented:
I like the new topic icons, but I'd like to keep the old 2 smileys... They're classics.

marc1706 commented:
What's the test file in develop for?

hanakin commented on tests/functional/fileupload_remote_test.php:
```diff
@@ -100,8 +100,8 @@ public function test_successful_upload()
 		$upload->set_error_prefix('')
 			->set_allowed_extensions(array('gif'))
-			->set_max_filesize(1000);
-		$file = $upload->handle_upload('files.types.remote', self::$root_url . 'styles/prosilver/theme/images/forum_read.gif');
```
@marc1706 because for some reason we are testing upload of the forum_read.gif to the files folder which does not make sense as this will no longer be a a file in 3.3 anyway so I just uploaded a random gif for performing these kind of tests to develop

hanakin commented:
@VSEphpbb we can always just bump the size
![mrgreen2](https://cloud.githubusercontent.com/assets/73081/22247011/a7f21f9e-e205-11e6-84a0-784f718f01c0.gif)

VSEphpbb commented:
Yeah, keep 'em but scale them up ;)

hanakin commented:
ok so the redface animation is not 100% the same i had to guess not having the actual layer style changes... Also I complete changed the thinking icon, but thoughts?

![fire](https://cloud.githubusercontent.com/assets/73081/22267826/3862ec3c-e253-11e6-9f35-6bb51cd20283.gif)
![heart](https://cloud.githubusercontent.com/assets/73081/22267827/3874efea-e253-11e6-8c9d-881e7f100bd3.gif)
![radioactive](https://cloud.githubusercontent.com/assets/73081/22267828/388821a0-e253-11e6-919b-52a1cf6a270a.gif)
![star](https://cloud.githubusercontent.com/assets/73081/22267829/3892832a-e253-11e6-8123-9f6b3099f5a4.gif)
![thinking](https://cloud.githubusercontent.com/assets/73081/22267830/38a46b44-e253-11e6-966c-9770fa816f9b.gif)
![alert](https://cloud.githubusercontent.com/assets/73081/22267831/38b297fa-e253-11e6-92b6-d530633caf30.gif)
![info](https://cloud.githubusercontent.com/assets/73081/22267832/38bee3a2-e253-11e6-800b-0fa8d9236c1a.gif)
![mrgreen](https://cloud.githubusercontent.com/assets/73081/22267833/38ce04fe-e253-11e6-9780-cd2d12cb515c.gif)
![question](https://cloud.githubusercontent.com/assets/73081/22267834/38ddf21a-e253-11e6-9910-26cb742a7178.gif)
![redface](https://cloud.githubusercontent.com/assets/73081/22267835/38f0c908-e253-11e6-9de9-9d5530256fc4.gif)

VSEphpbb commented:
i like them all

marc1706 commented:
Tests should be fixed now

hanakin commented:
@Nicofuma this should be good now

Note for release we should point out that to see the effect on custom themes they need to save 2x size versions of there icons

milescellar commented:
@hanakin @marc1706 these new retina imagesets looks really blurries on Chrome 59 (it's okay on Edge, and I don't know for other browsers). See: [http://imgur.com/a/TnmEm](http://imgur.com/a/TnmEm)

I tested on area51 and on a fresh phpBB 3.2.1-RC1 local installation. I cleared my entire browser cache, but the issue is still there.

hanakin commented:
@milescellar what is you OS and pls check to see if you graphics card drivers are up to date...I am assuming you are on a machine with a graphics card where you have access to update the drivers

milescellar commented:
@hanakin I'm on Windows 10 (up-to-date), and I'm using the latest graphics card drivers (I'm using an NVIDIA GeForce GTX 970 and I'm using the latest 384.76 GeForce drivers version). I don't think the graphic card drivers are the issue, I don't experienced any issue on other places on the Web or in games.

Crizz0 commented:
Is this maybe caused by the background-size-property?
```
dl.row-item {
    background-position: 10px 50%;
    background-repeat: no-repeat;
    background-size: 32px auto;
}
```

hanakin commented:
It's related to it as that is how you do retina! This is not the problem it is most likely caused by some weird render issue in this specific broweser on this OS possibly due to the use of gif vs png. Ultimately fixed in 3.3 with svgs. But we will have to test 3.2.1 for this specific situation. As I have not seen this problem on mac or 7 I did not test anything but edge on win 10

hanakin commented:
There is also a usecase for IE 11 that needs added to tests on win 10 as well as edge has a bad adoption rate still.

Noxwizard commented:
For me (Chrome 59, Win 10), adding `image-rendering: -webkit-optimize-contrast;` removed the blurriness.

hanakin commented:
Interesting? Will test when I get a chance? If works that is the best named property ever!

Scanialady commented:
> For me (Chrome 59, Win 10), adding image-rendering: -webkit-optimize-contrast; removed the blurriness.

-----

Where did you add it, please?

------
I like the new images.
Will you release a set of these images for dark styles, too? Would be helpful.

JimMH commented:
@Scanialady See this new pull request for that: https://github.com/phpbb/phpbb/pull/4880
